src/server/req_runjob.c

```diff
@@ -985,11 +985,15 @@ svr_strtjob2(job *pjob, struct batch_request *preq)
 
 	/* send the job to MOM */
 
+	/* clearing old comment since we are sending a fresh job */
+	job_attr_def[(int) JOB_ATR_Comment].at_decode(
+			&pjob->ji_wattr[(int) JOB_ATR_Comment], (char *) 0,
+			(char *) 0, "");
```

Comment:
you are right that we are not introducing a new message/comment but we are momentarily removing the comment from a job which is scheduled to run on a mom and is in substate 41. This wasn't the case earlier, So I think it does impact external behavior. Why don't we set the job comment to "Job run" (as it was earlier) or introduce a new "Job about to run" comment when server gets a run job request. This was we don't have to clear the comment and even mom rejects will show up on the job.